### PR TITLE
CXX-593: added bsoncxx::document::view::empty() helper

### PR DESCRIPTION
**Pull request by yhager**

### CODE REVIEW

File: src/bsoncxx/document/view.hpp [hunk 1]
```diff
@@ -121,6 +121,13 @@ class BSONCXX_API view {
     ///
     std::size_t length() const;
 
+    ///
+    /// Checks if the underlying document is empty
```
Comment (amidvidy) [inline src/bsoncxx/document/view.hpp]: I would elaborate a bit: 

```Checks if the underlying document is empty, i.e. it is equivalent to the trivial document '{}'.```

Also please put periods at the end of sentences.

**Comment by amidvidy** [comment]
Hi @yhager,

Thanks for the PR and apologies for the delayed reply as I was out of the office last week. This looks good to me, with a minor nit about the comment.

**Comment by acmorrow** [comment]
@amidvidy @yhager A question: What should happen in the view constructor if you pass NULL,0, or a pointer to valid bytes but length of 4?

**Comment by amidvidy** [comment]
The result of that would be undefined as the contract of the view constructor that takes a buffer/len is that the buffer contains *valid BSON*. Of course, the caller could use `bsoncxx::validate` to ensure this.

**Comment by acmorrow** [comment]
@amidvidy makes sense, thanks.

**Comment by yhager** [comment]
modified the comment as you requested. 
@acmorrow, AFAICT, a valid bson document cannot have 4 bytes.

**Comment by amidvidy** [comment]
merged in 05c1ca89. Thanks again @yhager!

**Pull request closed without merging**